packages-exp/auth-exp/src/core/credentials/email.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as externs from '@firebase/auth-types-exp';
+import { signInWithPassword } from '../../api/authentication/email_and_password';
+import { signInWithEmailLink } from '../../api/authentication/email_link';
+import { Auth } from '../../model/auth';
+import { IdTokenResponse } from '../../model/id_token';
+import { AuthErrorCode, AUTH_ERROR_FACTORY } from '../errors';
+import { EmailAuthProvider } from '../providers/email';
+import { debugFail } from '../util/assert';
+import { AuthCredential } from '.';
+
+export class EmailAuthCredential implements AuthCredential {
+  constructor(
+    readonly email: string,
+    readonly password: string,
+    readonly providerId: typeof EmailAuthProvider.PROVIDER_ID,
```

Comment:
why not just make this a readonly field and not have it on the constructor?

```
export class EmailAuthCredential implements AuthCredential {
  readonly providerId = EmailAuthProvider.PROVIDER_ID;
  constructor( ...
```

Reply:
done